gulp/createskin.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+
+var lazyReq = require('lazy-req')(require);
+var inquirer = require('inquirer');
+var fs = require('fs');
+var path = require('path');
+var through = lazyReq('through2');
+
+module.exports = function (gulp, gutil) {
+
+    /**
+     * This function should match all validations for studio skin creation. As of 16.4, studio skin creation does not
+     * allow underscores,spaces or dashes in skin name but allows them on the title.
+     */
+    function validate(value, regex, name, len) {
+        if (value === '') {
+            return 'Please provide a valid value';
+        }
+        var cases = value.match(regex);
+        if (cases === null) {
+            return 'Invalid value, please try again.';
+        }
+        if (value.length > len) {
+            return name + ' too long. Please use '+len+' chars or less';
+        }
+        return true;
+    }
+
+    /**
+     * Replace all spaces with underscores and converts the skin name to lowercase
+     */
+    function normalizeSkinId(value) {
+        if (value === '') {
+            throw Error("Bad input value");
+        }
+        return value.trim().replace(/ /g, '_').toLowerCase();
+    }
+
+    /**
+     * Exports core plugin skins for sdk if core plugins are not already downloaded.
+     */
+    function downloadCorePlugins() {
```

Comment:
I'm wondering if some of this logic should be moved down to plugin-export.js. It would be cool (out of scope for this change) to implement some hash check logic where we send a hash with the api and only return core plugin if it's changed, otherwise return "unchanged" http status code.